cylc/flow/job_runner_mgr.py

```diff
@@ -439,6 +439,12 @@ def _filter_submit_output(cls, st_file_path, job_runner, out, err):
     def _jobs_poll_status_files(self, job_log_root, job_log_dir):
         """Helper 1 for self.jobs_poll(job_log_root, job_log_dirs)."""
         ctx = JobPollContext(job_log_dir)
+        # If the log directory has been deleted prematurely, return a task
+        # failure and an explanation:
+        if not os.path.exists(os.path.join(job_log_root, ctx.job_log_dir)):
+            ctx.run_status = 1
+            ctx.run_signal = 'ERR/Job files have been removed'
```

Comment:
Should make this a constant to avoid duplicating this string in different places.

Reply:
It's only used once in `cylc/flow` - are you suggesting importing it into the tests?

Reply:
It's actually used twice in Cylc flow (but only once with the ERR/ prefix). Importing this into the jobs is a good idea.